fe/parameterize_batch.py

```diff
@@ -0,0 +1,523 @@
+
+            # set up the system
+            amber_ff = app.ForceField('amber99sbildn.xml', 'amber99_obc.xml')
+            host_system = amber_ff.createSystem(host_pdb.topology,
+                nonbondedMethod=app.NoCutoff,
+                constraints=None,
+                rigidWater=False)
+
+            host_system = openmm_converter.deserialize_system(host_system)
+            num_host_atoms = len(host_system.masses)
+
+            # open_ff = forcefield.Forcefield("ff/smirnoff_1.1.0.py")
+            open_ff = forcefield.Forcefield(args.forcefield)
+            nrg_fns = open_ff.parameterize(guest_mol)
+            guest_masses = get_masses(guest_mol)
+            guest_system = system.System(nrg_fns, open_ff.params, open_ff.param_groups, guest_masses)
+
+            combined_system = host_system.merge(guest_system)
+            cbs = -1*np.ones_like(np.array(combined_system.masses))*0.0001
+            lambda_idxs = np.zeros(len(combined_system.masses), dtype=np.int32)
+            lambda_idxs[num_host_atoms:] = -1
+
+            sim = simulation.Simulation(
+                combined_system,
+                step_sizes,
+                cas,
+                cbs,
+                lambda_schedule,
+                lambda_idxs,
+                precision
+            )
+
+            epoch_train_params = get_params(opt_state)
+
+            sim.system.params = epoch_train_params
+
+            all_args = []
+
+            child_conns = []
+            parent_conns = []
+            for conf_idx in range(num_conformers):
+
+                conformer = guest_mol.GetConformer(conf_idx)
+                guest_conf = np.array(conformer.GetPositions(), dtype=np.float64)
+                guest_conf = guest_conf/10 # convert to md_units
+                guest_conf = recenter(guest_conf, conf_com)
+                x0 = np.concatenate([host_conf, guest_conf])       # combined geometry
+
+                combined_pdb = Chem.CombineMols(Chem.MolFromPDBFile(host_pdb_file, removeHs=False), init_mol)
+                combined_pdb_str = StringIO(Chem.MolToPDBBlock(combined_pdb))
+                out_file = os.path.join(args.out_dir, "epoch_"+str(epoch)+"_insertion_deletion_"+host_name+"_conf_"+str(conf_idx)+".pdb")
+                writer = PDBWriter(combined_pdb_str, out_file)
+
+                v0 = np.zeros_like(x0)
+
+                parent_conn, child_conn = Pipe()
+                parent_conns.append(parent_conn)
+                # writer can be None if we don't care about vis
+                all_args.append([x0, v0, conf_idx % num_gpus, writer, child_conn])
+
+            processes = []
+
+            for arg in all_args:
+                p = Process(target=sim.run_forward_and_backward, args=arg)
+                p.daemon = True
+                processes.append(p)
+                p.start()
+
+            all_du_dls = []
+            for pc in parent_conns:
+                du_dls = pc.recv()
+                all_du_dls.append(du_dls)
+
```

Comment:
you need to close parent conns with pc.close() here else it will leak processes (run nvidia-smi and you should see memory usage start to creep up). You will also run out of fds on your os. 

I think you will also have to wrap the simulation.py code with:
``` python
simulation.py: run_forward_and_backward()

try:
        du_dl_adjoints = pipe.recv()
        stepper.set_du_dl_adjoint(du_dl_adjoints)
        ctxt.set_x_t_adjoint(np.zeros_like(x0))
        start = time.time()
        ctxt.backward_mode()
        print("bkwd run time", time.time() - start)
        dL_dp = ctxt.get_param_adjoint_accum()
        pipe.send(dL_dp)
except EOFError:
        pass
pipe.close()
return
```